src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hosting/ConfigurableLanguageServerFeatureOptions.cs

```diff
@@ -39,6 +39,7 @@ internal class ConfigurableLanguageServerFeatureOptions : LanguageServerFeatureO
     public override bool DisableRazorLanguageServer => _disableRazorLanguageServer ?? _defaults.DisableRazorLanguageServer;
     public override bool ForceRuntimeCodeGeneration => _forceRuntimeCodeGeneration ?? _defaults.ForceRuntimeCodeGeneration;
     public override bool UseNewFormattingEngine => _useNewFormattingEngine ?? _defaults.UseNewFormattingEngine;
+    public override bool SupportsSoftSelectionInCompletion => false;
```

Comment:
I very much don't like this. If we want to move away from configurability then we should move this class to rzls, and just hardcode everything that is not user configurable.

The purpose of this class was to allow the LSP server to support a range of methods of operating, and allowing the C# extension be in charge of which ones it opted in to. I have no problem with moving away from the design if that's what everyone thinks is best, but as they say in the movies, "not like this".

Reply:
Are you OK with leaving this as is in the PR and having a design conversation after you are back?

Reply:
> I have no problem with moving away from the design if that's what everyone thinks

I think it's best here but only because of 1 thing: This is basically a capabilities check that we're adding into our feature options. I'm more than happy to do the follow up work for leaving this configurable but having RZLS set this to false and other places setting it to true. I don't see a point in having the C# extension hardcode to true when in practice it's much easier to have the setting hardcodes all in one repo.

Reply:
> Are you OK with leaving this as is in the PR and having a design conversation after you are back?

You may rest assured I would have marked it "Request changes" if I wanted to actually block anything. While the cats away :)

Reply:
FWIW, this new (and better -- thanks!!) name indicates that this isn't really an option -- it's a client capability. It's the client communicating to the language server that it supports soft selection in completion. @davidwengier would pushing this into the client capabilities logic help alleviate your concern here?

Reply:
Indeed, I think an ideal fix here would be some system that can know what Html LSP server being used answered in its `initialize` response, so we can react to its capabilities as appropriate. I'm not aware of how we could do that, but I wouldn't be against it. I wouldn't be against much, my main complaint here is simply a class called "**Configurable**LanguageServerFeatureOptions" now has an option that is not configurable :P

(Yes I acknowledge that language server feature option is already a bit of a square hole that all manner of shapes of pegs have been fitted into)

Reply:
FWIW, my initial thought/suggestion here was to use SupportsVSExtensions from client capabilities. David's concern was that it might be not specific enough, thus a suggestion to use feature option flag that is more specific. Looks like we went full circle and are back to client capabilities, but want to define a new capability here?

Reply:
I _think_ you can check in as is. David said he would block if needed. We can discuss a potentially different approach in the future if needed. 

My opinion is that it SHOULD be a client capability but isn't. Using another one is odd instead of adding a new one. LanguageServerFeatureOptions is a way we can control it per client since they exist as per client implementations right now (sort of per client?). Current implementation solves the problem and doesn't cause in future known design problems so :shipit:  and get the fix to customers :)